panels/base_panel.py

```diff
@@ -338,6 +346,47 @@ def update_time(self):
             self.time_format = confopt
         return True
 
+    def battery_percentage(self, show=True):
```

Comment:
this is to big needs splitting and only updating if the ui elements are already in place, do not remove and recreate everything

Reply:
I split out the creation and i think improved the updating but if it needs to be split more let me know.